src/filesystem/impls/lib/handlers.js

```diff
@@ -0,0 +1,76 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
+/*global define */
+define(function (require, exports, module) {
+    "use strict";
+
+    var Content = require("filesystem/impls/lib/content");
+    var Log = require("filesystem/impls/lib/log");
+    var Rewriter = require("filesystem/impls/lib/rewriter");
+    var Filer = require("thirdparty/filer/dist/filer");
+    var Path  = Filer.Path;
+
+    function handle404(url, callback) {
```

Comment:
I wonder if we really want to do this.  Maybe we should start with 404 handling, and test it in the editor/preview, and see how it feels.  I kind of wonder if we instead just want to return errors on the callback...